pilot/pkg/networking/core/v1alpha3/cluster_builder.go

```diff
@@ -0,0 +1,146 @@
+// Copyright 2020 Istio Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha3
+
+import (
+	apiv2 "github.com/envoyproxy/go-control-plane/envoy/api/v2"
+	core "github.com/envoyproxy/go-control-plane/envoy/api/v2/core"
+	endpoint "github.com/envoyproxy/go-control-plane/envoy/api/v2/endpoint"
+
+	networking "istio.io/api/networking/v1alpha3"
+	"istio.io/istio/pilot/pkg/features"
+	"istio.io/istio/pilot/pkg/model"
+	"istio.io/istio/pilot/pkg/networking/util"
+	"istio.io/istio/pkg/config/labels"
+)
+
+var (
+	defaultDestinationRule = networking.DestinationRule{}
+)
+
+// applyDestinationRule applies the destination rule if it exists for the Service. It returns the subset clusters if any created as it
+// applies the destination rule.
+func applyDestinationRule(cluster *apiv2.Cluster, clusterMode ClusterMode, service *model.Service, port *model.Port, proxy *model.Proxy,
```

Comment:
I would suggest we make it as an interface of `ClusterBuilder` class to keep consistent and may extend more to it in the future

Reply:
Yeah. makes sense. Changed. PTAL

Reply:
Thanks